test/unittest/unittest_nntrainer_internal.cpp

```diff
@@ -773,20 +788,180 @@ TEST(nntrainer_DataBuffer, init_02_n) {
   nntrainer::DataBufferFromDataFile data_buffer;
   status = data_buffer.setMiniBatch(32);
   status = data_buffer.setClassNum(10);
-  EXPECT_EQ(status, ML_ERROR_NONE);  
-  status=data_buffer.setDataFile("trainingSet.dat", nntrainer::DATA_TRAIN);
-  EXPECT_EQ(status, ML_ERROR_NONE);  
-  status=data_buffer.setDataFile("valSet.dat",nntrainer::DATA_VAL);
-  EXPECT_EQ(status, ML_ERROR_NONE);  
-  status=data_buffer.setDataFile("testSet.dat",nntrainer::DATA_TEST);
-  EXPECT_EQ(status, ML_ERROR_NONE);  
-  status=data_buffer.setDataFile("label.dat", nntrainer::DATA_LABEL);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+  status = data_buffer.setDataFile("trainingSet.dat", nntrainer::DATA_TRAIN);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+  status = data_buffer.setDataFile("valSet.dat", nntrainer::DATA_VAL);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+  status = data_buffer.setDataFile("testSet.dat", nntrainer::DATA_TEST);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+  status = data_buffer.setDataFile("label.dat", nntrainer::DATA_LABEL);
   EXPECT_EQ(status, ML_ERROR_NONE);
   status = data_buffer.init();
   EXPECT_EQ(status, ML_ERROR_INVALID_PARAMETER);
 }
 
+/**
+ * @brief Data Buffer
+ */
+TEST(nntrainer_util_func, softmax_01_p) {
+  int status = ML_ERROR_NONE;
+  int batch = 3;
+  int height = 1;
+  int width = 10;
+  float results[10] = {7.80134161e-05, 2.12062451e-04, 5.76445508e-04,
+                       1.56694135e-03, 4.25938820e-03, 1.15782175e-02,
+                       3.14728583e-02, 8.55520989e-02, 2.32554716e-01,
+                       6.32149258e-01};
+
+  nntrainer::Tensor T(batch, height, width);
+  nntrainer::Tensor Results(batch, height, width);
+
+  for (int i = 0; i < batch; ++i) {
+    for (int j = 0; j < height; ++j) {
+      for (int k = 0; k < width; ++k) {
+        T.setValue(i, j, k, (i * (width) + k + 1));
```

Comment:
I think this can be also represented using `GEN_TEST_INPUT`
How about changing `GEN_TEST_INPUT` like
```
#define SET_TENSOR_DATA(T, eqation_i_j_k)  ... 
```

Reply:
fixed. Thanks.